conda/core/subdir_data.py

```diff
@@ -576,21 +512,55 @@ def maybe_decompress(filename, resp_content):
             """)
 
         else:
-            help_message = dals("""
-            An HTTP error occurred when trying to retrieve this URL.
-            HTTP errors are often intermittent, and a simple retry will get you on your way.
-            %s
-            """) % maybe_unquote(repr(e))
+            if url.startswith("https://repo.anaconda.com/"):
+                help_message = dals("""
+                An HTTP error occurred when trying to retrieve this URL.
+                HTTP errors are often intermittent, and a simple retry will get you on your way.
+
+                If your current network has https://www.anaconda.com blocked, please file
+                a support request with your network engineering team.
+
+                %s
+                """) % maybe_unquote(repr(e))
+            else:
+                help_message = dals("""
+                An HTTP error occurred when trying to retrieve this URL.
+                HTTP errors are often intermittent, and a simple retry will get you on your way.
+                %s
+                """) % maybe_unquote(repr(e))
 
         raise CondaHTTPError(help_message,
                              join_url(url, filename),
                              status_code,
                              getattr(e.response, 'reason', None),
                              getattr(e.response, 'elapsed', None),
-                             e.response)
-
-    except ValueError as e:
-        raise CondaIndexError("Invalid index file: {0}: {1}".format(join_url(url, filename), e))
+                             e.response,
+                             caused_by=e)
+
+    if resp.status_code == 304:
+        raise Response304ContentUnchanged()
+
+    def maybe_decompress(filename, resp_content):
+        return ensure_text_type(bz2.decompress(resp_content)
+                                if filename.endswith('.bz2')
+                                else resp_content).strip()
+
+    json_str = maybe_decompress(filename, resp.content)
+
+    saved_fields = {'_url': url}
+    add_http_value_to_dict(resp, 'Etag', saved_fields, '_etag')
+    add_http_value_to_dict(resp, 'Last-Modified', saved_fields, '_mod')
+    add_http_value_to_dict(resp, 'Cache-Control', saved_fields, '_cache_control')
+
+    # add extra values to the raw repodata json
+    if json_str and json_str != "{}":
+        raw_repodata_str = "%s, %s" % (
+            json.dumps(saved_fields)[:-1],  # remove trailing '}'
+            json_str[1:]  # remove first '{'
+        )
+    else:
+        raw_repodata_str = json.dumps(saved_fields)
+    return raw_repodata_str
```

Comment:
The above block of code was just moved out of the try/except block, to tighten up the error catching there.